unix/main.c

```diff
@@ -106,3 +106,30 @@ void uiQueueMain(void (*f)(void *data), void *data)
 	q->data = data;
 	gdk_threads_add_idle(doqueued, q);
 }
+
+struct timer {
+        int (*f)(void *);
+        void *data;
+};
+
+static gboolean dotimer(gpointer data)
+{
+        struct timer *t = (struct timer *) data;
+
+        if((*(t->f))(t->data))
+                return TRUE;
+        else {
+                g_free(t);
```

Comment:
I'd suggest using `uiNew()` and `uiFree()` for these, for consistency. (I now wonder if I need to keep these around anyway...)

Reply:
I vote we keep `uiNew()` and `uiFree()`. Very useful for debugging, especially across the different platforms. 

Also `uiQueueMain()` doesn't use them.. should it too? `uiTimer()` GTK+ implementation is based on it.

Reply:
`uiQueueMain()` doesn't use it because it is specifically intended for use across threads. Are you suggesting `uiTimer()` should also be used across threads? Currently `uiQueueMain()` is the only function allowed to be used by other threads, and `uiAlloc()` and friends are not thread safe.

Reply:
No no, you're right. Nevermind!

In fact maybe `uiQueueMain()` should be more clearly documented as the **only** function in the API that isn't UI thread only.